### DISCUSSION TITLE
Having issues with .filter(column.like(foo))

### DISCUSSION THREAD

**Discussion by crusty-dave** [diesel-rs/diesel#3050] [Q&A]
rustc 1.47.0
diesel 1.4.7

The following isn't working for me (it returns an empty vec):

```sql
CREATE TABLE my_table
(
    column1 Text NOT NULL,
    primary key (column1),
);
```

```rust

#[derive(Debug, Clone, Serialize, Deserialize, Insertable, Queryable, Identifiable, AsChangeset, AsExpression)]
#[table_name = "my_table"]
#[primary_key(column1)]
pub struct PgMyTable {
    pub(crate) column1 String,
}

            let foo = "foo".to_string;
            let query_result = my_table 
                .filter(column1.like(foo)
                .select(my_table::all_columns())
                .get_results::<PgMyTable>(conn);
```

However, the following returns all rows if I remove the filter:

```rust
            let query_result = my_table 
                .select(my_table::all_columns())
                .get_results::<PgMyTable>(conn);
```

The following also doesn't work:

```rust
            let foo = "foo".to_string.into_sql::<Text>();
            let query_result = my_table 
                .filter(column1.like(foo)
                .select(my_table::all_columns())
                .get_results::<PgMyTable>(conn);
```

If I run psql, the following is working:

```sql
SELECT * FROM my_table WHERE column1 LIKE 'foo';
```

The documentation on AsExpression is quite confusing, does foo in the above example somehow have to be given the trait AsExpression?  I haven't had any issues using .filter(column1.eq(foo), even when foo is &str.  However, like() & ilike() don't seem to work for me.

TIA, -Dave

**Comment by weiznich** [comment]
As with your previous questions: Please provide more information. This should at least include the relevant parts of your schema, the error message as emitted by rustc and the definitions of any involved  type/trait/.. that is not part of rust-std or diesel.

**Reply by crusty-dave** [reply]
There is no error message, I just get an empty vec returned.

I cannot provide an example of the schema as it stands, I was hoping that it might be an obvious issue with what I am doing.  If not, then I will need to spend some time to come up with a non-proprietary example.

For now, I will probably need to collect all of the rows, and then manually filter them in Rust code as I have already wasted a few hours on this...

Is there an example that shows how to use filter(column.like()) somewhere?  I just need to see a working example and I will probably be able to figure it out.

TIA, -Dave

**Reply by weiznich** [reply]
The API [documentation contains](https://docs.diesel.rs/1.4.x/diesel/expression_methods/trait.TextExpressionMethods.html#examples) an example. 

 > I cannot provide an example of the schema as it stands, I was hoping that it might be an obvious issue with what I am doing. If not, then I will need to spend some time to come up with a non-proprietary example.

Generally speaking: I'm not willing to spend my time on reverse engineering proprietary in my free time to answer questions. So either you are able to somehow provide the corresponding or equivalent code or we are not able to provide support for such issues here. In the second case you can consider a paid consultant contract.

**Reply by crusty-dave** [reply]
@weiznich  I never expected you to.  I thought that given this is a discussion board, that someone else may have had the same issue and might chime in.  I will probably create an example to demonstrate the issue at some point. Like I said, I was just hoping for something obvious.

Sorry to have bothered you, thanks for taking the time to respond.

**Comment by crusty-dave** [comment]
It turned out to be a stupid bug on my part, probably a result of working too many hours without enough rest.

The code that I had above actually worked once I found / fixed it.

```rust
            let foo = "foo".to_string;
            let query_result = my_table 
                .filter(column1.like(foo)
                .select(my_table::all_columns())
                .get_results::<PgMyTable>(conn);
```

Sorry, -Dave
